### PR TITLE
[zh] Sync Advancements in Kubernetes Traffic Engineering blog into Chinese

### PR DESCRIPTION
**Pull request by wilsonwu**
Translate Sync Advancements in Kubernetes Traffic Engineering blog into Chinese:
```
content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md
```

### CODE REVIEW

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+Prior to Kubernetes v1.26, clusters could experience [loss of traffic](https://github.com/kubernetes/kubernetes/issues/85643) from Service load balancers during rolling updates when setting the `externalTrafficPolicy` field to `Local`. There are a lot of moving parts at play here so a quick overview of how Kubernetes manages load balancers might help!
+-->
+在 Kubernetes v1.26 之前，将 `externalTrafficPolicy` 字段设置为 `Local` 时，
+集群在滚动更新期间可能会遇到 Service 的负载均衡器[流量丢失](https://github.com/kubernetes/kubernetes/issues/85643)问题。
+这里有很多活动部件作用其中，因此简述 Kubernetes 管理负载均衡器的机制可能对此有所帮助！
+
+<!--
+In Kubernetes, you can create a Service with `type: LoadBalancer` to expose an application externally with a load balancer. The load balancer implementation varies between clusters and platforms, but the Service provides a generic abstraction representing the load balancer that is consistent across all Kubernetes installations.
+-->
+在 Kubernetes 中，您可以创建一个 `type: LoadBalancer` 的 Service，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md:44]: 译文中不用尊称“您”，统一用“你”。

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+并通过负载均衡器对外暴露应用程序。负载均衡器的实现因集群和平台而异，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md:45]: application 统一翻译为"应用"，不要翻译为“应用程序”

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+Under the hood, Kubernetes allocates a NodePort for the Service, which is then used by kube-proxy to provide a network data path from the NodePort to the Pod. A controller will then add all available Nodes in the cluster to the load balancer’s backend pool, using the designated NodePort for the Service as the backend target port.
+-->
+在底层，Kubernetes 为 Service 分配一个 NodePort，然后 kube-proxy 使用它来提供从
+NodePort 到 Pod 的网络数据路径。然后，控制器将集群中的所有可用节点添加到负载均衡器的后端池中，
+使用 Service 的指定 NodePort 作为后端目标端口。
+
+<!--
+{{< figure src="traffic-engineering-service-load-balancer.png" caption="Figure 1: Overview of Service load balancers" >}}
+-->
+{{< figure src="traffic-engineering-service-load-balancer.png" caption="图 1：Service 负载均衡器概述" >}}
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md:73]:
```suggestion
{{< figure src="traffic-engineering-service-load-balancer.png" caption="图 1：Service 负载均衡器概览" >}}
```

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+<!--
+Oftentimes it is beneficial to set `externalTrafficPolicy: Local` for Services, to avoid extra hops between Nodes that are not running healthy Pods backing that Service. When using `externalTrafficPolicy: Local`, an additional NodePort is allocated for health checking purposes, such that Nodes that do not contain healthy Pods are excluded from the backend pool for a load balancer.
+-->
+通常，为 Service 设置 `externalTrafficPolicy: Local` 是有益的，
+用于避免在该 Service 背后的健康 Pod 未运行时节点之间出现额外的跳跃。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md:79]:
```suggestion
对于没有运行为该 Service 提供服务的健康 Pod 的节点而言，可以避免在这类节点之间的额外跳转。
```

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+{{< figure src="traffic-engineering-lb-without-proxy-terminating-endpoints.png" caption="图 3：当 externalTrafficPolicy 为 Local 时，从负载均衡器到正在终止的端点的流量" >}}
+
+<!--
+Starting in Kubernetes v1.26, kube-proxy enables the `ProxyTerminatingEndpoints` feature by default, which adds automatic failover and routing to terminating endpoints in scenarios where the traffic would otherwise be dropped. More specifically, when there is a rolling update and a Node only contains terminating Pods, kube-proxy will route traffic to the terminating Pods based on their readiness. In addition, kube-proxy will actively fail the health check NodePort if there are only terminating Pods available. By doing so, kube-proxy alerts the external load balancer that new connections should not be sent to that Node but will gracefully handle requests for existing connections.
+-->
+从 Kubernetes v1.26 开始，kube-proxy 将默认启用 `ProxyTerminatingEndpoints` 功能，
+该功能在流量将被丢弃时自动添加故障转移并将流量路由到正在终止的端点。更具体地说，
+当存在滚动更新并且节点仅包含正在终止的 Pod 时，kube-proxy 会根据其就绪情况将流量路由到正在终止的 Pod 中。
+此外，如果只有正在终止的 Pod 可用时，kube-proxy 将主动使 NodePort 的健康检查失败。
+通过这样做，kube-proxy 会提醒外部负载均衡器新连接不应发送到该节点，并妥善处理现有连接的请求。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md:109]:
```suggestion
通过这种做法，kube-proxy 能够提醒外部负载均衡器新连接不应发送到该节点，并妥善处理现有连接的请求。
```

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+## 欢迎参与 {#getting-involved}
+
+<!--
+If you're interested in future discussions on Kubernetes traffic engineering, you can get involved in SIG Network through the following ways:
+* Slack: [#sig-network](https://kubernetes.slack.com/messages/sig-network)
+* [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-network)
+* [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/sig%2Fnetwork)
+* [Biweekly meetings](https://github.com/kubernetes/community/tree/master/sig-network#meetings)
+-->
+如果您对未来关于 Kubernetes 流量工程的讨论感兴趣，可以通过以下方式参与 SIG Network：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md:208]:
```suggestion
如果你对未来关于 Kubernetes 流量工程的讨论感兴趣，可以通过以下方式参与 SIG Network：
```

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md (hunk 1 above)
Comment (wilsonwu) [reply content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md]: Fixed all.

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md (hunk 2 above)
Comment (wilsonwu) [reply content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md]: Fixed

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md (hunk 3 above)
Comment (wilsonwu) [reply content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md]: Applied

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md (hunk 4 above)
Comment (wilsonwu) [reply content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md]: Applied

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md (hunk 5 above)
Comment (wilsonwu) [reply content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md]: Applied

File: content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md (hunk 6 above)
Comment (wilsonwu) [reply content/zh-cn/blog/_posts/2022-12-30-advancements-in-traffic-engineering/index.md]: Fixed

**Pull request merged**